File: services/telemetry-service/src/main/java/net/blueshell/telemetry/service/TelemetryService.java
package net.blueshell.telemetry.service;

import net.blueshell.enums.PlatformType;
import net.blueshell.db.BaseModelService;
import net.blueshell.telemetry.model.Telemetry;
import net.blueshell.telemetry.repository.TelemetryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.transaction.annotation.Transactional;

import java.util.UUID;

@org.springframework.stereotype.Service
public class TelemetryService extends BaseModelService<Telemetry, UUID, TelemetryRepository> {

    @Autowired
    public TelemetryService(TelemetryRepository repository) {
        super(repository);
    }

    @Transactional
    public Telemetry createTelemetry(PlatformType platform, String url) {
        Telemetry telemetry = new Telemetry(platform, url);
        create(telemetry);
        return telemetry;
    }
}

File: api/src/main/kotlin/net/blueshell/api/domain/telemetry/application/TelemetryService.kt
package net.blueshell.api.domain.telemetry.application

import net.blueshell.api.shared.enums.PlatformType
import net.blueshell.api.shared.service.BaseModelService
import net.blueshell.api.telemetry.persistence.Telemetry
import net.blueshell.api.telemetry.persistence.repository.TelemetryRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class TelemetryService @Autowired constructor(repository: TelemetryRepository, events: ApplicationEventPublisher) :
    BaseModelService<Telemetry, Long, TelemetryRepository>(repository) {
    @Transactional
    fun createTelemetry(platform: PlatformType, url: String): Telemetry {
        val telemetry = Telemetry(platform, url)
        create(telemetry)
        return telemetry
    }
}
